### PR TITLE
remove panos_security_policy

### PR DESCRIPTION
**Pull request by ivanbojer**
##### SUMMARY
removed panos_security_policy deprecated module in lieu of new panos_security_rule module. The old module uses deprecated API's that do not exist anymore.

##### ISSUE TYPE
 - Feature Pull Request

##### COMPONENT NAME
panos_security_policy

##### ANSIBLE VERSION
```
ansible 2.4.2.0
  config file = None
  configured module search path = [u'/Users/ibojer/.ansible/plugins/modules', u'/usr/share/ansible/plugins/modules']
  ansible python module location = /Users/ibojer/.virtualenvs/ansible/lib/python2.7/site-packages/ansible
  executable location = /Users/ibojer/.virtualenvs/ansible/bin/ansible
  python version = 2.7.13 (default, Feb  6 2017, 11:22:10) [GCC 4.2.1 Compatible Apple LLVM 8.0.0 (clang-800.0.42.1)]
```

### CODE REVIEW

**Comment by vinayvenkat** [comment]
Ship it!

**Comment by ivanbojer** [comment]
ready_for_review

**Comment by vinayvenkat** [comment]
shipit

**Comment by shinmog** [comment]
shipit

**Comment by gundalow** [comment]
Hi,
Need to go through a deprecation cycle for this. Can't just remove the modules

See what was done in http://docs.ansible.com/ansible/devel/dev_guide/developing_modules_documenting.html

https://github.com/ansible/ansible/pull/34100

1) Give notice of deprecation
`git mv` file so it starts with `_`
Add `DOCUMENATION.deprecation` dict
Update changelog to give noticed, under the "will be deleted in Ansible 2.9" section
Add entry into `porting_guide_2.5.rst
2) Wait till Ansible 2.9
Change module to be docs only

**Review by gundalow — CHANGES_REQUESTED** [review]
Hi,
Need to go through a deprecation cycle for this. Can't just remove the modules

See what was done in http://docs.ansible.com/ansible/devel/dev_guide/developing_modules_documenting.html

https://github.com/ansible/ansible/pull/34100

1) Give notice of deprecation
`git mv` file so it starts with `_`
Add `DOCUMENATION.deprecation` dict
Update changelog to give noticed, under the "will be deleted in Ansible 2.9" section
Add entry into `porting_guide_2.5.rst
2) Wait till Ansible 2.9
Change module to be docs only

**Comment by ivanbojer** [comment]
I've addressed the comments. If this is correct I will do the same for the other module. @gundalow thank you for the feedback and please let me know if it looks good now.

File: lib/ansible/modules/network/panos/_panos_security_policy.py [hunk 1]
```diff
@@ -0,0 +1,519 @@
...
+module: panos_security_policy
+short_description: Create security rule policy on PanOS devices.
+description:
+    - Security policies allow you to enforce rules and take action, and can be as
+      general or specific as needed. The policy rules are compared against the
+      incoming traffic in sequence, and because the first rule that matches the
+      traffic is applied, the more specific rules must precede the more general ones.
+author: "Ivan Bojer (@ivanbojer)"
+version_added: "2.3"
+deprecated: In 2.4 use M(panos_security_rule) instead.
```
Comment (gundalow) [inline lib/ansible/modules/network/panos/_panos_security_policy.py]:
```
deprecated:
    removed_in: 2.9
    why: Uses deprecated API calls
    alternative: Use M(panos_security_rule) instead.
``

**Review by gundalow — CHANGES_REQUESTED** [review]

File: lib/ansible/modules/network/panos/_panos_security_policy.py [hunk 2]
```diff
@@ -35,8 +35,9 @@
       traffic is applied, the more specific rules must precede the more general ones.
 author: "Ivan Bojer (@ivanbojer)"
 version_added: "2.3"
+deprecated: In 2.4 use M(panos_security_rule) instead.
 deprecated:
-  removed_in: "2.8"
+  removed_in: "2.9"
```
Comment (gundalow) [inline lib/ansible/modules/network/panos/_panos_security_policy.py]: Missing `alternative:`

File: lib/ansible/modules/network/panos/_panos_security_policy.py [hunk 3]
```diff
@@ -36,9 +36,11 @@
 author: "Ivan Bojer (@ivanbojer)"
 version_added: "2.3"
 deprecated:
-  removed_in: "2.8"
-  why: Renamed to M(panos_security_rule) in order to align with API calls and UI object references, which also has extra support for PanDevice SDK.
-  alternative: Use M(panos_security_rule) instead.
+    alternative:
```
Comment (gundalow) [inline lib/ansible/modules/network/panos/_panos_security_policy.py]: `alternative: Use M(panos_security_rule) instead.`

`why: This module depended on outdated and old SDK.`

**Comment by ivanbojer** [comment]
@gundalow I think this might do it 👍

**Comment by ivanbojer** [comment]
ready_for_review

**Review by gundalow — APPROVED** [review]

**Pull request merged**